src/distilabel/pipeline/base.py

```diff
@@ -189,7 +188,7 @@ def run(
         self,
         parameters: Optional[Dict[str, Dict[str, Any]]] = None,
         use_cache: bool = True,
-    ) -> "Distiset":  # type: ignore
+    ) -> None:
```

Comment:
This was like this on purpose, to denote that classes extending this method must return a `Distiset`.

Reply:
Hmm then shouldn't that be a protected / private method as that's just called via `super()` and instead create an `abstractmethod` for the `run` method?

Reply:
I can revert it for the moment, but that felt off to me :/

Reply:
We can create a `_run` abstractmethod that can be later be invoked by `run`, but I think that's adding complexity which is not needed. The docstring also specified that this method is meant to be extended.